Storybook tables

BartDeCaluwe opened the pull request:
This PR will add:

- Pagniation component
- Tab component with corresponding TabPane component
- TableHeaderCell component. This component allows us to easily add icons to table headers
- Table component
- Snapshot tests for Pagination, Tab and Table components
- Storybook stories for Pagination, Tab and Table components

BartDeCaluwe committed:
feat: add pagination component

BartDeCaluwe committed:
feat: add TableHeaderCell component

BartDeCaluwe committed:
feat: add Table component

BartDeCaluwe committed:
feat: add Tab component

BouweCeunen commented:
small conflict it seems

BartDeCaluwe committed:
test: add missing icon snapshots

RuvendeGroote commented on src/stories/Table.stories.tsx:
```diff
@@ -0,0 +1,166 @@
+	],
+	columns: [
+		{
+			title: () => <TableHeaderCell label="Name" icon="stacked-column-down" />,
```
what's your opinion on hiding/having to do this?
we can have our own column object definition that has title and optional icon and then perform a map in the table component itself replacing those with what you are doing here?

RuvendeGroote replied:
I want a clean simple api and I feel like this is just something you have to know/think about but it's just default behaviour. we can always allow a function to be passed to let the user render their own header like ant design allows

BartDeCaluwe replied:
Yes, that's definitely something we can do. I agree that it'd make things easier, as a tableheader without icon would probably be the exception rather than the rule.

RuvendeGroote commented on src/stories/Table.stories.tsx:
```diff
@@ -0,0 +1,166 @@
+	],
+};
+
+export const WithTabs = Template.bind({});
```
imo, this shouldn't be part of the table stories as it's not part of the table component, it's a combination of components, but I guess it works for demonstration purposes.

BartDeCaluwe replied:
Yeah, I wasn't sure either. I included it because it's specifically mentioned in the designs.

BartDeCaluwe committed:
Refactor: add optional icon and titleClassname to column

This way we can avoid having to pass around a render function every time we want to render a table heading with an icon.
